SourceCode/raytracer/CRTTypes.h

```diff
@@ -0,0 +1,209 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <stdexcept>
+#include <cmath>
+
+struct Pixel
+{
+    uint8_t r = 0;
+    uint8_t g = 0;
+    uint8_t b = 0;
+    std::string toString() const
+    {
+        return std::to_string(r) + " " + std::to_string(g) + " " + std::to_string(b) + "\t";
+    }
+};
+
+class Buffer
+{
+public :
+    Buffer(uint16_t _width, uint16_t _height) : width(_width), height(_height)
+    {
+        data = std::make_unique<Pixel[]>(width * height);
+    }
+
+    std::unique_ptr<Pixel[]> data;
+
+    const uint16_t width;
+    const uint16_t height;
+};
+
+class Vec3 {
+public:
+    float x, y, z;
+
+    Vec3() : x(0), y(0), z(0) {}
```

Comment:
I prefer using the {} constructor, here you're using integer literals to initialize floats {} constructor will warn you about that.